android/src/main/java/org/reactnative/camera/tasks/BarCodeScannerAsyncTask.java

```diff
@@ -2,139 +2,66 @@
-  private int mWidth;
-  private int mHeight;
+  private int mDataWidth;
+  private int mDataHeight;
   private BarCodeScannerAsyncTaskDelegate mDelegate;
   private final MultiFormatReader mMultiFormatReader;
-  private boolean mLimitScanArea;
-  private float mScanAreaX;
-  private float mScanAreaY;
-  private float mScanAreaWidth;
-  private float mScanAreaHeight;
-  private int mCameraViewWidth;
-  private int mCameraViewHeight;
-  private float mRatio;
+  private Rect mRect;
+  private boolean mAcceptableRect;
+  private String mOutOfBoundErrorMessage = "BarCodeScannerAsyncTask mRect dimensions are OutOfBounds, mRect set to Full Screen";
 
-  //  note(sjchmiela): From my short research it's ok to ignore rotation of the image.
   public BarCodeScannerAsyncTask(
       BarCodeScannerAsyncTaskDelegate delegate,
       MultiFormatReader multiFormatReader,
       byte[] imageData,
-      int width,
-      int height,
-      boolean limitScanArea,
-      float scanAreaX,
-      float scanAreaY,
-      float scanAreaWidth,
-      float scanAreaHeight,
-      int cameraViewWidth,
-      int cameraViewHeight,
-      float ratio
+      int dataWidth,
+      int dataHeight,
+      Rect rect
   ) {
+    mDataWidth = dataWidth;
+    mDataHeight = dataHeight;
     mImageData = imageData;
-    mWidth = width;
-    mHeight = height;
     mDelegate = delegate;
     mMultiFormatReader = multiFormatReader;
-    mLimitScanArea = limitScanArea;
-    mScanAreaX = scanAreaX;
-    mScanAreaY = scanAreaY;
-    mScanAreaWidth = scanAreaWidth;
-    mScanAreaHeight = scanAreaHeight;
-    mCameraViewWidth = cameraViewWidth;
-    mCameraViewHeight = cameraViewHeight;
-    mRatio = ratio;
+    Rect defaultRect = new Rect(0, 0, mDataWidth, mDataHeight);
+    mAcceptableRect = defaultRect.contains(rect) && !rect.isEmpty();
+    mRect = mAcceptableRect ? rect : defaultRect;
   }
 
   @Override
   protected Result doInBackground(Void... ignored) {
-    if (isCancelled() || mDelegate == null) {
-      return null;
-    }
-
+    if (isCancelled() || mDelegate == null) { return null; }
+    if (!mAcceptableRect) { Log.w("CAMERA_1::", mOutOfBoundErrorMessage); }
     Result result = null;
-    /**
-     * mCameraViewWidth and mCameraViewHeight are obtained from portait orientation
-     * mWidth and mHeight are measured with landscape orientation with Home button to the right
-     * adjustedCamViewWidth is the adjusted width fromt the Aspect ratio setting 
-     */
-    int adjustedCamViewWidth = (int) (mCameraViewHeight / mRatio);
-    float adjustedScanY = (((adjustedCamViewWidth - mCameraViewWidth) / 2) + (mScanAreaY * mCameraViewWidth)) / adjustedCamViewWidth;
-    
-    int left = (int) (mScanAreaX * mWidth);
-    int top = (int) (adjustedScanY * mHeight);
-    int scanWidth = (int) (mScanAreaWidth * mWidth);
-    int scanHeight = (int) (((mScanAreaHeight * mCameraViewWidth) / adjustedCamViewWidth) * mHeight);
-
+    boolean landscapeMode = mDataWidth > mDataHeight;
```

Comment:
I forgot to unlock my phone orientation settings and I will have to make some modification.